### PR TITLE
New CRT marketplace query

### PR DESCRIPTION
**Pull request by ikprk**

### CODE REVIEW

File: src/server-extension/resolvers/CreatorToken/index.ts [hunk 1]
```diff
@@ -41,6 +59,291 @@ export class TokenResolver {
...
+      const channelsMapped = oldListQMap(rows)
+      return channelsMapped.map((creatorToken, i) => ({
+        creatorToken,
+        ammVolume: ammVolumes[i] ?? 0,
+      }))
+    }
+
+    const result = await ctx.openreader.executeQuery(listQuery)
+
+    return result as TokenReturnType[]
```
Comment (zeeshanakram3) [inline src/server-extension/resolvers/CreatorToken/index.ts:144]: Incorrect type? `ammVolume` is not a field in `TokenReturnType `

File: src/server-extension/resolvers/CreatorToken/index.ts [hunk 2]
```diff
@@ -41,6 +59,291 @@ export class TokenResolver {
...
+    const topTokensCtes = `
+WITH  tokens_volumes AS (
+   SELECT ac.token_id,
+        SUM(tr.price_paid) as ammVolume
+   FROM amm_transaction tr
+   JOIN amm_curve ac ON ac.id = tr.amm_id
+   WHERE tr.created_in >= ${lastProcessedBlock - args.periodDays * BLOCKS_PER_DAY}
+   GROUP BY token_id
+)
+`
```
Comment (zeeshanakram3) [inline src/server-extension/resolvers/CreatorToken/index.ts:91]:
```suggestion
    const topTokensCtes = `
      WITH tokens_volumes AS (
          SELECT
              ac.token_id,
              SUM(tr.price_paid) as ammVolume
          FROM
              amm_transaction tr
              JOIN amm_curve ac ON ac.id = tr.amm_id
          WHERE
              tr.created_in >= $ { lastProcessedBlock - args.periodDays * BLOCKS_PER_DAY }
          GROUP BY
              token_id
      )
    `
```

**Review by zeeshanakram3 — CHANGES_REQUESTED** [review]
Please run `npm run generate:docs:auth-api` auth API docs

File: src/mappings/token/index.ts [hunk 3]
```diff
@@ -233,7 +233,8 @@ export async function processAmmActivatedEvent({
     ammInitPrice: BigInt(intercept),
     finalized: false,
   })
-  token.lastPrice = amm.ammInitPrice
+  token.lastPrice =
+    (amm.ammSlopeParameter * BigInt(token.totalSupply)) / BigInt(2) + amm.ammInitPrice
```
[the remark this answers is not in the document]
Comment (ikprk) [reply src/mappings/token/index.ts:237]: tbh I copied it over from Atlas, Ignazio knows the details of this calc

File: src/auth-server/handlers/registerUserInteraction.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        .where('entity.entityId = :entityId', { entityId })
+        .andWhere('entity.type  = :type', { type })
+        .andWhere('entity.dayTimestamp >= :startOfDay', { startOfDay })
+        .andWhere('entity.dayTimestamp <= :endOfDay', { endOfDay })
+        .getOne()
+
+      if (!dailyInteractionRow) {
+        await em.getRepository(UserInteractionCount).save({
+          id: `${Date.now()}-${entityId}-${type}`,
+          dayTimestamp: new Date(),
```
[the remark this answers is not in the document]
Comment (ikprk) [reply src/auth-server/handlers/registerUserInteraction.ts:61]: I think we match by day only in the resolver, but it seems like a good improvement anyway

File: src/server-extension/resolvers/StateResolver/index.ts [hunk 5]
```diff
@@ -74,6 +79,28 @@ export class StateResolver {
     return state
   }
 
+  @Query(() => [TopInteractedEntity])
+  async getTopInteractedEnities(
+    @Args() args: TopInteractedEntityArgs
+  ): Promise<TopInteractedEntity[]> {
+    const em = await this.tx()
+
+    const result: { entity_id: string; entrycount: number }[] = await em.query(`
```
[the remark this answers is not in the document]
Comment (ikprk) [reply src/server-extension/resolvers/StateResolver/index.ts:88]: Hmmm, it does wont correctly, so I wouldn't change it

File: src/server-extension/resolvers/StateResolver/types.ts [hunk 6]
```diff
@@ -1,7 +1,37 @@
...
+  entityId: string
+
+  @Field({ nullable: false })
+  interactionCount: number
+}
+
+@ArgsType()
+export class TopInteractedEntityArgs {
+  @Field(() => String, { nullable: false })
+  type: string
```
[the remark this answers is not in the document]
Comment (ikprk) [reply src/server-extension/resolvers/StateResolver/types.ts:21]: Hmmm, good question. On one side there might be a scenario where we would like to use this logic in more that one place, maybe with new type, so each change would require Orion changes as well. Do you see any security problem here? If not, maybe we can leave it as it is, and Atlas  shall be responsible for making sure to use some common variable to reference certain `type`.

[the remark this answers is not in the document]
Comment (ikprk) [reply src/server-extension/resolvers/StateResolver/types.ts:21]: @dmtrjsg do you think we will have more logic like hot tokens on marketplace that will require us to count some channel, video or whatever entity inside Atlas to show the most clicked ones?

File: src/server-extension/resolvers/StateResolver/index.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (zeeshanakram3) [reply src/server-extension/resolvers/StateResolver/index.ts:88]: you mean work?

File: src/server-extension/resolvers/CreatorToken/index.ts [hunk 7]
```diff
@@ -41,6 +58,296 @@ export class TokenResolver {
...
+      model,
+      ctx.openreader.dialect,
+      'MarketplaceToken',
+      marketplaceTokensFields,
+      sqlArgs
+    )
+
+    let listQuerySql = listQuery.sql
+
+    listQuerySql = listQuerySql.replace(/marketplace_token/g, 'marketplace_tokens')
```
Comment (zeeshanakram3) [inline src/server-extension/resolvers/CreatorToken/index.ts:343]: This should not be required any longer?

**Pull request merged**